src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,209 @@
+        }
+    }
+
+    @Override
+    public List<Car> getCarByDriver(Long driverId) {
+        String findCarsQuery = "SELECT c.id AS car_id, model, manufacturer_id "
+                + " FROM cars c JOIN cars_drivers cd ON c.id = cd.car_id WHERE "
+                + "cd.driver_id = ? AND c.is_deleted = FALSE;";
+        List<Car> allCars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarsStatement = connection.prepareStatement(findCarsQuery)) {
+            getCarsStatement.setLong(1, driverId);
+            ResultSet resultSet = getCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                allCars.add(getCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't find cars by driver id: " + driverId, e);
+        }
+        return allCars;
+    }
+
+    private Car getCarFromResultSet(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject(1, Long.class);
+        String model = resultSet.getString("model");
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        Manufacturer manufacturer = manufacturerDao.get(manufacturerId).get();
+        Car car = new Car();
+        car.setId(id);
+        car.setModel(model);
+        car.setManufacturer(manufacturer);
+        car.setDrivers(getDriversForCar(id));
+        return car;
+    }
+
+    private List<Driver> getDriversForCar(Long carId) {
+        String getAllDriversQuery = "SELECT d.id AS driver_id, d.name AS driver_name, "
+                + "d.license_number AS driver_license_number FROM drivers d JOIN cars_drivers  "
+                + "cd ON d.id = cd.driver_id WHERE cd.car_id = ? AND d.is_deleted = FALSE;";
+        List<Driver> allDrivers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllDriversStatement = connection
+                        .prepareStatement(getAllDriversQuery)) {
+            getAllDriversStatement.setLong(1, carId);
+            ResultSet resultSet = getAllDriversStatement.executeQuery();
+            while (resultSet.next()) {
+                allDrivers.add(getDriverFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't find drivers by car id " + carId, e);
+        }
+        return allDrivers;
+    }
+
+    private Driver getDriverFromResultSet(ResultSet resultSet) throws SQLException {
+        Driver driver = new Driver();
+        driver.setId(resultSet.getObject("driver_id", Long.class));
+        driver.setName(resultSet.getString("driver_name"));
+        driver.setLicenseNumber(resultSet.getString("driver_license_number"));
+        return driver;
+    }
+
+    private void setDriverCarRelation(Driver driver, Car car) {
```

Comment:
```suggestion
    private void insertDrivers(Car car) {
```
we don't need to pass single driver, use list that is the field of car

Reply:
Done